public/ledger.js

```diff
@@ -0,0 +1,290 @@
+    }
+
+    if (result.length === 8 && result.slice(0, 3).equals(Buffer.from(constants.ledgerVersionMagicStringBuffer))) {
+      this.hathorAppOpened = true;
+      return result;
+    } else {
+      throw new Error('Invalid Hathor app.');
+    }
+  }
+
+  /**
+   * Get public key data that will be used to generate the xpub
+   *
+   * @return {Promise} Promise resolved with the data as parameter. Might be rejected in case of an error or if the user denies access.
+   */
+  getPublicKeyData = async () => {
+    try {
+      const transport = await this.getTransport();
+      return await this.sendToLedgerOrQueue(transport, this.commands.PUBLIC_KEY_DATA, 0, 0);
+    } catch (e) {
+      throw Ledger.parseLedgerError(e);
+    }
+  }
+
+  /**
+   * Send address command so ledger can show specific address for user to validate
+   *
+   * @param {Buffer} index Address index to be shown on ledger
+   *
+   * @return {Promise} Promise resolved when user validates that the address is correct.
+   */
+  checkAddress = async (index) => {
+    try {
+      const transport = await this.getTransport();
+      const result = await this.sendToLedgerOrQueue(transport, this.commands.ADDRESS, 0, 0, index);
+      return result;
+    } catch (e) {
+      throw Ledger.parseLedgerError(e);
+    }
+  }
+
+  sendRequest = (transport, command, p1, p2, data) => {
+    return transport.send(0xe0, command, p1, p2, data);
+  }
+
+  /**
+   * Send transaction to ledger
+   *
+   * Maximum data that can be transferred to ledger is 255 bytes
+   *
+   * p1 = 0
+   * p2 = remaining calls
+   * Eg: 3 rounds (data has 612 bytes)
+   * p1 p2 data
+   * 0  2  255 bytes
+   * 0  1  255 bytes
+   * 0  0  102 bytes
+   *
+   * @param {Object} data Data to send to Ledger (change_output_info + sighash_all)
+   *
+   * @return {Promise} Promise resolved when data is received
+   */
+  sendTx = async (data) => {
+    let offset = 0;
+    try {
+      const transport = await this.getTransport();
+      while (offset < data.length) {
+        const toSend = data.slice(offset, offset + maxLedgerBuffer);
+        await this.sendToLedgerOrQueue(transport, this.commands.SEND_TX, 0, 0, toSend);
+        offset += maxLedgerBuffer;
+      }
+    } catch (e) {
+      throw Ledger.parseLedgerError(e);
+    }
+  }
+
+  /**
+   * Get signatures
+   *
+   * We request the signature for each input
+   *
+   * p1 = 1
+   * p2 = *
+   * data = priv key index (4 bytes)
+   * ## done, no more signatures needed
+   * p1 = 2
+   * p2 = *
+   * data = none
+   *
+   * @param {Object} indexes Index of the key we want the data signed with
+   *
+   * @return {Promise} Promise resolved when all signatures are received. Promise is resolved with an array with the signatures.
+   */
+  getSignatures = async (indexes) => {
+    const transport = await this.getTransport();
+    return await this.handleGetSignatures(transport, indexes);
+  }
+
+  handleGetSignatures = async (transport, indexes) => {
```

Comment:
Missing docstring.

Reply:
Merged this method with the `getSignatures`. There was no reason for having both.